activiti-core/activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessRuntimeCandidateStartersIT.java

```diff
@@ -0,0 +1,137 @@
+        Page<ProcessDefinition> processDefinitionPage = processRuntime.processDefinitions(Pageable.of(0,
+                                                                                                      50));
+        assertThat(processDefinitionPage.getContent()).isNotNull();
+        assertThat(processDefinitionPage.getContent()).hasSize(1);
+        assertThat(processDefinitionPage.getContent().get(0).getKey()).isEqualTo(RESTRICTED_PROCESS_DEFINITION_KEY);
+    }
+
+    @Test
+    public void candidateStarterGroupMembers_should_getProcessDefinitions() {
+        securityUtil.logInAs("john");
+
+        Page<ProcessDefinition> processDefinitionPage = processRuntime.processDefinitions(Pageable.of(0,
+            50));
+        assertThat(processDefinitionPage.getContent()).isNotNull();
+        assertThat(processDefinitionPage.getContent()).hasSize(1);
+        assertThat(processDefinitionPage.getContent().get(0).getKey()).isEqualTo(RESTRICTED_PROCESS_DEFINITION_KEY);
+    }
+
+    @Test
+    public void nonCandidateStarters_shouldNot_getProcessDefinitions() {
+        securityUtil.logInAs("garth");
+
+        Page<ProcessDefinition> processDefinitionPage = processRuntime.processDefinitions(Pageable.of(0,
+            50));
+        assertThat(processDefinitionPage.getContent()).isNotNull();
+        assertThat(processDefinitionPage.getContent()).hasSize(0);
+    }
+
+
+    @Test
+    public void candidateStarterUser_should_getProcessDefinition() {
+        securityUtil.logInAs("user");
+
+        ProcessDefinition processDefinition = processRuntime.processDefinition(RESTRICTED_PROCESS_DEFINITION_KEY);
+        assertThat(processDefinition).isNotNull();
+        assertThat(processDefinition.getKey()).isEqualTo(RESTRICTED_PROCESS_DEFINITION_KEY);
+    }
+
+    @Test
+    public void candidateStarterGroupMembers_should_getProcessDefinition() {
+        securityUtil.logInAs("john");
+
+        ProcessDefinition processDefinition = processRuntime.processDefinition(RESTRICTED_PROCESS_DEFINITION_KEY);
+        assertThat(processDefinition).isNotNull();
+        assertThat(processDefinition.getKey()).isEqualTo(RESTRICTED_PROCESS_DEFINITION_KEY);
+    }
+
+    @Test
+    public void nonCandidateStarters_shouldNot_getProcessDefinition() {
+        securityUtil.logInAs("garth");
+
+        Throwable throwable = catchThrowable(() -> processRuntime.processDefinition(RESTRICTED_PROCESS_DEFINITION_KEY));
+
+        assertThat(throwable)
+            .isInstanceOf(ActivitiObjectNotFoundException.class)
+            .hasMessage("Unable to find process definition for the given id:'" + RESTRICTED_PROCESS_DEFINITION_KEY + "'");
+    }
+
+    @Test
+    public void candidateStarterUser_can_startProcess() {
+        securityUtil.logInAs("user");
+
+        Page<ProcessInstance> processInstancePage = processRuntime.processInstances(PAGEABLE);
+
+        assertThat(processInstancePage).isNotNull();
+        assertThat(processInstancePage.getContent()).isEmpty();
```

Comment:
I don't think this check is needed: checking if the `start` method is returning a valid process instance is enough.

Reply:
I was making sure there are no previously existing running processes. But you're right, there shouldn't be any running processes instances because they are deleted after each test.